SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllerTests.cs

```diff
@@ -0,0 +1,290 @@
+
+            var controller = new GiftController(mockService.Object);
+
+            //act
+            var getResult = await controller.Get(42);
+
+            // Assert
+            Assert.IsNull(getResult.Value);
+
+
+
+        }
+
+        [TestMethod]
+        public async Task PutById_WithNewGift_Success()
+        {
+
+            // Arrange
+
+            var mockService = new Mock<IGiftService>();
+
+            Gift gift = SampleData.CreateGift();
+
+            mockService.Setup(s => s.UpdateAsync(42, gift))
+                .ReturnsAsync(gift);
+
+            var controller = new GiftController(mockService.Object);
+
+            //act
+            ActionResult<Gift> putResult = await controller.Put(42, gift);
+
+
+            // Assert
+            Assert.IsTrue(putResult.Result is OkResult);
+        }
+
+        [TestMethod]
+        public async Task PutById_GiftNotFoud_ReturnsNull()
+        {
+
+            // Arrange
+
+            var mockService = new Mock<IGiftService>();
+
+            Gift gift = SampleData.CreateGift();
+
+            Gift nullGift = null!;
+
+            mockService.Setup(s => s.UpdateAsync(42, nullGift))
+                .ReturnsAsync(nullGift);
+
+            var controller = new GiftController(mockService.Object);
+
+
+            //act
+            var putResult = await controller.Put(42, gift);
+
+            // Assert
+            Assert.IsTrue(putResult.Result is NotFoundResult);
+        }
+
+        [TestMethod]
+        public async Task Delete_EntityWithId_Success()
+        {
+            //Arrange
+            var mockService = new Mock<IGiftService>();
+
+
+            mockService.Setup(s => s.DeleteAsync(42)).ReturnsAsync(true);
+
+
+            var controller = new GiftController(mockService.Object);
+
+            //Act
+            ActionResult deleteResult = await controller.Delete(42);
+
+            //Assert
+            Assert.IsTrue(deleteResult is OkResult);
+        }
+
+        [TestMethod]
+        public async Task Delete_GiftWithWrongId_Fails()
+        {
+            //Arrange
+            var mockService = new Mock<IGiftService>();
+
+
+            mockService.Setup(s => s.DeleteAsync(42)).ReturnsAsync(false);
+
+
+            var controller = new GiftController(mockService.Object);
+
+            //Act
+            ActionResult deleteResult = await controller.Delete(42);
+
+            //Assert
+            Assert.IsTrue(deleteResult is NotFoundResult);
+        }
+
+
+        /*  [HttpGet]
+          public async Task<IEnumerable<TEntity>> Get()
+          {
+              List<TEntity> entities = await EntityService.FetchAllAsync();
+              return entities;
+          }
+  */
+        [TestMethod]
+        public async Task Get_WithGifts_ReturnsAllGifts()
+        {
+            //whats the difference between why have to use IGiftService? versus GiftService
+            // Arrange
+           List<Gift> gifts = new List<Gift>();
```

Comment:
You can use a collection initializer to make this simpler.
```suggestion
           List<Gift> gifts = new List<Gift>
           {
               SampleData.CreateGift(),
               SampleData.CreateGift()
           };
```